packages/rpc-types/src/lamports.ts

```diff
@@ -46,18 +49,45 @@ export function lamports(putativeLamports: bigint): LamportsUnsafeBeyond2Pow53Mi
     return putativeLamports;
 }
 
+type ExtractAdditionalProps<T, U> = Omit<T, keyof U>;
+
 export function getLamportsEncoder(): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
     return getMemoizedU64Encoder();
 }
 
+export function getLamportsEncoderFromEncoder<TEncoder extends NumberEncoder>(
```

Comment:
nit: How do you feel about making this the main `getLamportsEncoder` function, and call this other one something like `getU64LamportsEncoder` instead?

Reply:
I don't have a strong view either way, but I'd prefer to do it in a separate PR just to reduce churn and files here. I probably already should have split this one a bit!

Reply:
Sounds good!